src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -193,11 +261,26 @@ def _validate_vm_create_vnet(namespace):
             # 2 - user specified existing vnet/subnet
             namespace.vnet_type = 'existing'
             return
-
     # 3 - create a new vnet/subnet
     namespace.vnet_type = 'new'
 
 
+def _validate_vmss_create_subnet(namespace):
+    # TODO: we can consider for non-new sceanrio there are user asks
+    if namespace.vnet_type == 'new':
+        if namespace.subnet_address_prefix is None:
+            cidr = namespace.vnet_address_prefix.split('/', 1)[0]
+            i = 0
+            for i in range(8, 32, 1):
```

Comment:
What is the purpose of this logic here? Is this a port of the prefix defaulting logic from vnet create?